sotodlib/core/metadata/obsdb.py

```diff
@@ -81,10 +101,38 @@ def __init__(self, map_file=None, init_db=True):
                     changes = True
             if changes:
                 self.conn.commit()
-
+        self.primary_keys = self.get_primary_keys()
+        if ~np.isin(len(self.primary_keys), [1,3]):
+            raise ValueError("Unsupported number of primary keys in the 'obs' table"
+                             " only 1 or 3 are allowed.")
+
+    def get_primary_keys(self):
+        """Retrieve the primary keys of the specified table.
+           This is used whether to index by obs_id or obs-wafer-band."""
+        query = "PRAGMA table_info('obs')"
+        c = self.conn.execute(query)
+        primary_keys = [row['name'] for row in c.fetchall() if row['pk'] > 0]
+        return primary_keys
+    
+    def warn_primary_keys(self, wafer_band):
+        """Warn the user if the primary keys are not specified 
+           and we're defaulting to using _all."""
+        if (None in wafer_band):
+            warn_str = 'WARNING: Primary key(s)'
+            for i, wb in enumerate(wafer_band):
+                if wb is None:
+                    wafer_band[i] = '_all'
```

Comment:
This failed for me because I passed in a tuple.  Also it's not great to modify someone's inputs if you don't need to.  Instead, start the func with `wafer_band = list(wafer_band)`; then you'll definitely be able to change it and return it.